Makefile

```
@@ -552,13 +557,20 @@ docker-build: ## Build the docker image for controller-manager
 docker-push: ## Push the docker image
 	docker push $(CONTROLLER_IMG)-$(ARCH):$(TAG)
 
-.PHONY: build-fkas
+.PHONY: docker-build-fkas
 # Allow overriding this by setting CONTAINER_RUNTIME var
 CONTAINER_RUNTIME := $(if $(CONTAINER_RUNTIME),$(CONTAINER_RUNTIME),docker)
 export CONTAINER_RUNTIME
 
-build-fkas:
-	cd $(FAKE_APISERVER_DIR) && $(CONTAINER_RUNTIME) build --build-arg ARCH=$(ARCH) -t "quay.io/metal3-io/metal3-fkas:latest" .
+docker-build-fkas:
+	cp -r $(FAKE_APISERVER_DIR) /tmp && \
+	mkdir -p /tmp/fake-apiserver/capm3  && \
+	cp -r ./api /tmp/fake-apiserver/capm3 && \
+	cd /tmp/fake-apiserver && \
+	$(GO) mod edit -replace=github.com/metal3-io/cluster-api-provider-metal3=./capm3 && \
+	$(GO) mod tidy && \
+	$(CONTAINER_RUNTIME) build --build-arg ARCH=$(ARCH) -t "quay.io/metal3-io/metal3-fkas:latest" . || true
+	sudo rm -rf /tmp/fake-apiserver
```

```suggestion
	rm -rf /tmp/fake-apiserver
```
I don't see anything in the temp dir that would require sudo.